### ISSUE TITLE
Required label in push row should be removed if value is present

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36608]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.42-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:**  https://expensify.slack.com/archives/C049HHMV9SM/p1708010208553859

## Action Performed:
1. go to workspace chat 
2. Click + and request money
3. go to distance
4. Add start and stop point
5. Click continue
6. Observe the category field

## Expected Result:
"Required" shouldn't appear if already a value present
## Actual Result:
The required label persists even after the value is there.
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Screen Shot 2024-02-15 at 11 29 27 AM](https://github.com/Expensify/App/assets/38435837/2d3c4789-2013-4f00-b378-cd281335009b)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Required label shows even though the value is present for the push row

### What is the root cause of that problem?
At the moment we do not have any condition to not show the required label when the value for the push item is present.

### What changes do you think we should make in order to solve the problem?
I can see we used `rightLabel` only for showing the `required` value for the MenuItems, so to generic solution we can consider the `title` value to prevent rendering the `rightLabel`.
```js
!title && !!rightLabel
```
https://github.com/Expensify/App/blob/2fe384c5a2279d65f083d828f825dc9368a3f2ef/src/components/MenuItem.tsx#L605-L609

<details>
<summary>Result</summary>

https://github.com/Expensify/App/assets/14358475/18f7f517-3a84-466c-9221-5c63ba5b8f91

</details>

**Comment by Pujan92** [comment]
@aimane-chnaif @cead22 I have updated my [proposal](https://github.com/Expensify/App/issues/36608#issuecomment-1946675790), Can you plz give a look at it as I think it helps to reduce the conditions and be more generic.

**Comment by Pujan92** [comment]
Ok, I thought it makes sense to add it as it defines the intended behavior. Bcoz for right side content we already have `rightComponent` prop.

**Comment by cead22** [comment]
I like @Pujan92 's proposal better because all the required labels work the same today in that if there's a value to display (`title`) in the menu item, we shouldn't show the required label. This is kinda similar to the approach that we took in this issue https://github.com/Expensify/App/issues/35843#issuecomment-1944591292, where the component has all the logic it needs to decide whether to display something or not, and therefore, the logic should be inside the component and not outside of it (like with `!iouCategory && isCategoryRequired ? translate('common.required') : ''`)

We wouldn't be introducing a new prop, we'd be replacing `rightLabel` with `showRequiredLabel`.

If we want to keep the change a bit smaller, and keep the ability of putting something other than `Required` on the right label, we can keep the `rightLabel` as is, and do something like this on `MenuItem.tsx`
```js
{!title &&  (
    <View style={styles.justifyContentCenter}>
        <Text style={styles.rightLabelMenuItem}>{rightLabel}</Text>
    </View>
)}
```

**Comment by Pujan92** [comment]
@aimane-chnaif @cead22 To confirm, do we need to proceed by updating the prop to `showRequiredLabel` or keep the `rightLabel` and use `title` value to render?

**Comment by cead22** [comment]
Yeah. Let's keep the `rightLabel` and use `title` value to render?

**Comment by Pujan92** [comment]
Thanks, [PR](https://github.com/Expensify/App/pull/37137) is ready for review!

**Comment by cead22** [comment]
This on the payment stage, so not overdue

**Comment by muttmuure** [comment]
Was ooo, handling now

**Comment by muttmuure** [comment]
All paid up

**Comment by muttmuure** [comment]
@aimane-chnaif @Pujan92 could you confirm if we need a regression test?
